Unity-Technologies/ScriptableRenderPipeline
# Catch and handle NRefactory parser exceptions

### Purpose of this PR
Context:  https://fogbugz.unity3d.com/f/cases/1225164/

The tool for generating HLSL structs from tagged C# structs relies on NRefactory, which has not been maintained in 5 years.  Some of our newer code uses C# 6.0 syntax (out var, ref var, etc), which isn't supported by NRefactory, which causes the tool to throw unhandled exceptions when certain packages are imported (collections, entities, etc).  The proper "fix" for this would be to port the code to use Roslyn instead, but that's a somewhat significant undertaking.

This PR adds code to catch NRefactory parser exceptions and report the filename.

---
### Testing status

**Manual Tests**: What did you do?
Verified that the tool recovers from parsing errors (see the repro steps in the Fogbugz report).

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline
